src/openforms/formio/api/serializers.py

```diff
@@ -44,3 +46,24 @@ def get_url(self, instance) -> str:
             kwargs={"uuid": instance.uuid},
             request=request,
         )
+
+    def validate_file(self, value: UploadedFile):
+        if value.content_type == "application/octet-stream":
+            ext = value.name.split(".")[-1]
```

Comment:
we prefer the modern `pathlib.Path(value.name).suffix`

Reply:
That's what I had initially. I <3 pathlib, but though it is semantically incorrect, it's not really a path, just a name. i.e. checking on `.exists()` leads to drawing incorrect conclusions about the state of the file etc.

Reply:
fair enough. I suppose you could argue that it's a relative path ( :wink: ) but I'm not married to it.

I also did a quick check for the content type lookup (e.g. with `.tar.gz` file where you'd have multiple suffixes), but in the end only the last bit is relevant (this would yield a `gzip` content type, tar does not have one for itself) which is also in line with expectations.

Reply:
lol, relative to what? The `django.core.file.uploadedfile.UploadedFile` can be a proxy to memory, a cdn, anything really. I went with `PurePath.suffix` as a compromise :handshake:

magic allows for inspection into these types like zipfiles and tarballs. Should the check in `openforms.submissions.attachments.validate_uploads` check if all the contents of these archive files are allowed too?